monash-human-power/MHP_DAS
# Change to new topics and sensors

## Description

**Most important changes**
+ Converted topics into an enum file for easier readability and decreased hardcoding
+ Fix changed topics eg. /v3/wireless-module/<id>/low-battery. Now all of the code matches the spec on notion
+ Added new sensors (power, cadence and heartRate)
+ Added support for multiple mock sensors (any number can now be specified eg. 20)
+ Removed huge if statements to define the modules, now it is simply a function and you can specify what sensors you want in a list

**Less important changes** 
+ Fixed some documentation so now it makes sense
+ If statements now check the entire topic rather than just the start and the end
+ `if None` hack to specify all sensors has been removed so now all the sensors that are being mocked are just stored in an array
+ Saved file names are now generated by the logging script in the format `module_id + "_D%d-%m-%d_T%H-%M-%S" + ".csv"`

## Screenshots

<img width="365" alt="image" src="https://user-images.githubusercontent.com/23159604/75339846-672d5280-58e5-11ea-8f65-5ba723380f3e.png">
As many sensors as you need / want

## Steps to Test
1. Make sure mosquitto is running
2. Start logger `python3 mqtt_wireless_log.py`
3. Start mock modules `python3 mock_wireless_module.py -t 5 --id 1 2 3 4 5 666`

Watch as 6 fake modules generate and save data for 5 seconds
